sirepo/simulation_db.py

```diff
@@ -764,6 +764,16 @@ def user_path(uid=None, check=False):
     return d
 
 
+def validate_path(path_parts, uid):
+    with sirepo.auth.set_user(uid):
+        # TODO(e-carlin): handle more than sim files
+        assert '/'.join(path_parts[:-1]) in str(simulation_dir(path_parts[2], path_parts[3])), \
+            f'invaid path={path_parts}'
+        f = path_parts[-1]
```

Comment:
The number of parts was not checked explicitly but it was checked. We verified that the user supplied path minus the filename was contained within the known simulation dir path. Your path did not pass.

I've implemented the regex 15c9a9c8. I don't believe we can define the regex at startup because we need to verify that the uid we got from the auth header is the same uid in the path. Not just that it passes a regex for something that looks like a uid.

I've added a test. 

Thoughts?